src/test/java/io/openliberty/tools/intellij/it/GradleSingleModMPSIDProjectTest.java

```diff
@@ -0,0 +1,252 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.tools.intellij.it;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * Tests Liberty Tools actions using a single module MicroProfile Gradle project with space in directory and name.
+ */
+public class GradleSingleModMPSIDProjectTest extends SingleModMPProjectTestCommon {
+
+    /**
+     * Single module Microprofile project name.
+     */
+    private static final String SM_MP_PROJECT_NAME = "singleModGradleMP";
+
+    /**
+     * Single module Microprofile project name with space.
+     */
+    private static final String SM_MP_PROJECT_NAME_NEW = "singleMod GradleMP";
+
+    /**
+     * The path to the folder containing the test projects.
+     */
+    private static final String PROJECTS_PATH = Paths.get("src", "test", "resources", "projects", "gradle").toAbsolutePath().toString();
+
+    /**
+     * The path to the folder containing the test projects, including directories with spaces.
+     */
+    private static final String PROJECTS_PATH_NEW = Paths.get("src", "test", "resources", "projects", "gradle sample").toAbsolutePath().toString();
+
+    /**
+     * Project port.
+     */
+    private final int SM_MP_PROJECT_PORT = 9080;
+
+    /**
+     * Project resource URI.
+     */
+    private final String SM_MP_PROJECT_RES_URI = "api/resource";
+
+    /**
+     * Project response.
+     */
+    private final String SM_MP_PROJECT_OUTPUT = "Hello! Welcome to Open Liberty";
+
+    /**
+     * Relative location of the WLP installation.
+     */
+    private final String WLP_INSTALL_PATH = "build";
+
+    /**
+     * The path to the test report.
+     */
+    private final Path TEST_REPORT_PATH = Paths.get(PROJECTS_PATH_NEW, SM_MP_PROJECT_NAME_NEW, "build", "reports", "tests", "test", "index.html");
+
+    /**
+     * Build file name.
+     */
+    private final String BUILD_FILE_NAME = "build.gradle";
+
+    /**
+     * Action command to open the build file.
+     */
+    private final String BUILD_FILE_OPEN_CMD = "Liberty: View Gradle config";
+
+    /**
+     * Dev mode configuration start parameters.
+     */
+    private final String DEV_MODE_START_PARAMS = "--hotTests";
+
+    /**
+     * Dev mode configuration custom start parameters for debugging.
+     */
+    private final String DEV_MODE_START_PARAMS_DEBUG = "--libertyDebugPort=9876";
+
+    /**
+     * Prepares the environment for test execution.
+     */
+    @BeforeAll
+    public static void setup() throws IOException {
+        // Copy the directory from PROJECTS_PATH to PROJECTS_PATH_NEW
+        TestUtils.copyDirectory(PROJECTS_PATH, PROJECTS_PATH_NEW);
```

Comment:
If you handle the IOException from this line you can remove the `throws` definition above (in combination with the changes below.

Reply:
If an exception occurs within the `setup()` method, it will result in an initialization error because the method will not execute `prepareEnv(PROJECTS_PATH_NEW, SM_MP_PROJECT_NAME_NEW`). Consequently, all tests in the parent class `SingleModMPProjectTestCommon` will not work, leading to an` initialization error`.